lib/hwm/hwm.go

```diff
@@ -120,6 +139,15 @@ func (trie *HighWatermarkTracker) DeleteOldKeys(hwmLimit uint64) []string {
 	return deletedKeys
 }
 
+func (trie *HighWatermarkTracker) ToMap() map[string]uint64 {
```

Comment:
Alternative is to put the tests in the same package and poke into private state.  I think this is a bit cleaner.